tests/test_EEE_module/test_emissions.py

```diff
@@ -0,0 +1,426 @@
+        )
+    ],
+)
+def test_calculate_purchased_feed_emissions(
+    homegrown_feeds: list[dict[str, Any]],
+    purchased_feeds: dict[str, float],
+    actual_purchased_feeds: dict[str, float],
+    actual_purchased_feed_emissions: dict[str, float],
+    actual_land_use_change_emissions: dict[str, float],
+    mocker: MockerFixture,
+) -> None:
+    """Tests the calculation of purchased feed emissions."""
+    em = EmissionsEstimator()
+    mock_add = mocker.patch.object(OutputManager, "add_variable")
+    mock_gather_feeds = mocker.patch.object(em, "_gather_ration_feed_totals", return_value=purchased_feeds)
+    mock_calc_actual = mocker.patch.object(em, "_calculate_actual_purchased_feeds", return_value=actual_purchased_feeds)
+    mock_calc_actual_emission = mocker.patch.object(
+        em,
+        "_calculate_actual_purchased_feed_emissions",
+        return_value=(actual_purchased_feed_emissions, actual_land_use_change_emissions),
+    )
+
+    em._calculate_purchased_feed_emissions(homegrown_feeds)
+
+    assert mock_add.call_count == 3
+    mock_gather_feeds.assert_called_once()
+    mock_calc_actual.assert_called_once_with(homegrown_feeds, purchased_feeds)
+    mock_calc_actual_emission.assert_called_once_with(actual_purchased_feeds)
+
+
+def test_gather_homegrown_feeds_and_fertilizer_apps(mocker: MockerFixture) -> None:
+    """Tests that the homegrown feeds and fertilizer applications were gathered correctly."""
+    em = EmissionsEstimator()
+    mock_filter_variable = mocker.patch.object(
+        em.om,
+        "filter_variables_pool",
+        return_value={"Time.day": {"values": [20]}, "Time.calendar_year": {"values": [2014]}},
+    )
+    mock_convert_time = mocker.patch.object(Time, "convert_year_jday_to_date", return_value=datetime(2014, 1, 20))
+    mock_filter_results = mocker.patch.object(em, "_filter_results", return_value=[{"dry_yield": 4, "field_size": 2}])
+    time_filter = {
+        "name": "Time Filter",
+        "description": "Collects the date a year before the simulation ended, to be used as a cutoff for deciding "
+        "which crop yields and nutrient applications to estimate emissions for.",
+        "filters": ["Time.(day|calendar_year)"],
+        "slice_start": -365,
+        "slice_end": -364,
+    }
+
+    expected = {
+        "Homegrown Feeds": [{"dry_yield": 4, "field_size": 2, "total_dry_yield": 8}],
+        "Fertilizer Applications": [{"dry_yield": 4, "field_size": 2, "total_dry_yield": 8}],
+        "Manure Applications": [{"dry_yield": 4, "field_size": 2, "total_dry_yield": 8}],
+        "Manure Requests": [{"dry_yield": 4, "field_size": 2, "total_dry_yield": 8}],
+    }
+    observed = em._gather_homegrown_feeds_and_fertilizer_apps()
+
+    assert observed == expected
+    mock_convert_time.assert_called_once_with(2014, 20)
+    mock_filter_variable.assert_called_once_with(time_filter)
+    assert mock_filter_results.call_count == 4
+    mock_filter_results.assert_has_calls(
```

Comment:
```suggestion
    mock_filter_results.assert_has_calls(
```
The call count is checked implicitly when `assert_has_calls` is called.